generate_release_notes.py

```diff
@@ -80,33 +98,39 @@ def get_parent_epic(jira_issue, jira_url, jira_email, jira_token):
     return None, None, None
 
 def get_issues_from_pr(github_repo, pr_number):
+    """
+    Get Jira keys and GitHub issues from the pull request body.
+
+    Args:
+        github_repo (Repository): The GitHub repository object.
+        pr_number (int): The number of the pull request.
+        jira_key (str): The Jira issue key (example, 'C1T-1234').
+
+    Returns:
+        tuple: Lists of Jira keys and GitHub issues.
+    """
     github_pull_request = github_repo.get_pull(pr_number)
-    # Remove <!-- comments --> from issue body
     try:
-        issue_body = re.sub('<!--.*-->', '', github_pull_request.body)
-    except:
+        issue_body = re.sub(r'<!--.*-->', '', github_pull_request.body)
+    except TypeError:
         issue_body = ""
-        pass
 
     jira_keys = []
     github_issues = []
 
     if issue_body:
-        # Get Jira keys only from the "Related Jira Key" section of PR's
-        jira_key_match = re.findall(r'Related Jira Key.*?(\[.*?\])?\(?([A-Z]+-\d+)\)?', issue_body, re.DOTALL)
+        jira_key_match = re.findall(r'Related Jira Key.*?(\[[^\]]*\])?\(?([A-Z]+-\d+)\)?', issue_body, re.DOTALL)
         if jira_key_match:
             jira_keys = [match[1].strip() for match in jira_key_match if match[1].strip()]
 
-        # If no Jira keys found, check for GitHub Issues from the "Related GitHub Issue" section
         if not jira_keys:
-            github_issue_match = re.findall(r'Related GitHub Issue.*?(\[.*?\])?\(?#(\d+)\)?', issue_body, re.DOTALL)
+            github_issue_match = re.findall(r'Related GitHub Issue.*?(\[[^\]]*\])?\(?#(\d+)\)?', issue_body, re.DOTALL)
```

Comment:
Why is this regex different from the one above?

Reply:
I changed this regex accordinglg from pylint warnings as takes only characters inside the square brackets which are matched but it extracts GitHub issue numbers from the pull request body with no effect.